### PR TITLE
Move HTML2PDF to modules directory

### PR DESCRIPTION
**Pull request by tmccormi**
Removing this component from another project to /modules.
This module is called from one location in the code:
/interface/patient_file/report/custom_report.php

### CODE REVIEW

**Comment by aethelwulffe** [comment]
bump

**Comment by tmccormi** [comment]
just to be clear, it's just a move, not a massive code change

File: interface/patient_file/report/custom_report.php [hunk 1]
```diff
@@ -46,7 +46,7 @@
 $PDF_OUTPUT = empty($_POST['pdf']) ? 0 : intval($_POST['pdf']);
 
 if ($PDF_OUTPUT) {
-  require_once("$srcdir/html2pdf/vendor/autoload.php");
+  require_once("../../../modules/html2pdf/vendor/autoload.php");
```
Comment (tmccormi) [inline interface/patient_file/report/custom_report.php]: backwards relative paths ... someday we need a modules location constant ..

**Comment by teryhill** [comment]
It appears to just be a move. what do you want to call the modules global?

**Comment by aethelwulffe** [comment]
How about $modules ?

**Comment by aethelwulffe** [comment]
Or $modules_root or $modules_dir
..I like those better

**Comment by teryhill** [comment]
$globals['modules_dir']

**Comment by aethelwulffe** [comment]
Added $GLOBALS['modules_dir'] to the branch.  This will break it, and will only be testable when the additional global is added.  I am doing this under the assumption that this is a reviewed requirement, and not something that we were going to change after committing.

**Comment by teryhill** [comment]
@tmccormi  It appears to be a move . I up voted this

**Comment by aethelwulffe** [comment]
I will be marking in the new globals, but those are in the nation notes branch at the moment.  It would be a second commit merge to do those first, then four more commits to the two branches in question...so I would like to just test this again as-is, then if the nation notes gets merged, I can just add one more commit to this to add the global paths.  This does not require a toggle.  I am mostly doing it to make integrating the PDF functions in the whole application to one module we can maintain and use it's documentation to track everywhere it is used (and needs test of hooks when changed).  Then the call to the PDF modules can be added to the API we will have one day.

File: interface/patient_file/report/custom_report.php [hunk 2]
```diff
@@ -46,7 +46,7 @@
 $PDF_OUTPUT = empty($_POST['pdf']) ? 0 : intval($_POST['pdf']);
 
 if ($PDF_OUTPUT) {
-  require_once("$srcdir/html2pdf/vendor/autoload.php");
+  require_once($GLOBALS['modules_dir'] . "/html2pdf/vendor/autoload.php");
   $pdf = new HTML2PDF ($GLOBALS['pdf_layout'],
```
Comment (teryhill) [inline interface/patient_file/report/custom_report.php]: I did a search and this is used in sl_eob_search.php also.

Comment (aethelwulffe) [inline interface/patient_file/report/custom_report.php]: Uh, I only see one of the other PDF functions there, which is class.ezpdf.php which is an extension of class.pdf.php, both of which are still in /library/classes

Comment (aethelwulffe) [inline interface/patient_file/report/custom_report.php]: Am I wrong?  "HTML2" does not show up in that file...

Comment (teryhill) [inline interface/patient_file/report/custom_report.php]: The change to add Ray's statements had that change. Is it not still in the Codebase?

Comment (teryhill) [inline interface/patient_file/report/custom_report.php]: line 94 of sl_eob_search.php

**Comment by aethelwulffe** [comment]
Line 94 of interface/billing/sl_eob_search.php reads:    while(!feof($file))
Apparently either Ray's changes never made it in, or perhaps it could be that my branch is crap because it has been sitting here for months.
  I am not re-building, re-basing this or anything like it.  If it takes a second commit to make the sl_eob bit work, fine.  I am investing way too much time working on my stale pull requests.

**Comment by teryhill** [comment]
I checked and the code is in the main code base.

**Comment by aethelwulffe** [comment]
You are right.  sl_eob has changed since then, and this feature was added.

**Comment by aethelwulffe** [comment]
Should I just burn this?

**Comment by teryhill** [comment]
no commit it and then fix sl_eob on another commit,

**Comment by aethelwulffe** [comment]
It isn't cool to commit your own pulls.  Red five standing by.

**Comment by teryhill** [comment]
Where is Blue Leader @tmccormi

**Comment by aethelwulffe** [comment]
I know where Gold Leader is....

**Comment by tmccormi** [comment]
blueleader is going to trust the team on this one ...

**Pull request merged**